Exerscise_3/main.cpp

```diff
@@ -70,7 +71,7 @@ std::ifstream read_from_file(const std::string &file) {
 
 
 int main(int args, char **argv){
-    if(args==2) {
+    if(args>=2) {
         std::vector<std::pair<int, std::unique_ptr<IRecord>>> cource_vector;
         std::vector<std::pair<int, std::unique_ptr<IRecord>>> exam_vector;
         std::vector<std::pair<int, std::unique_ptr<IRecord>>> teacher_vector;
```

Comment:
Like I said, you could use `std::map<int, std::unique_ptr<>>` here. It'd sort the entries for you.